examples/siro_sandbox/sandbox_app.py

```diff
@@ -1032,12 +1038,9 @@ def sim_update(self, dt):
         post_sim_update_dict: Dict[str, Any] = {}
 
         if self.gui_input.get_key_down(GuiInput.KeyNS.ESC):
+            self._check_save_episode_data()
```

Comment:
Save recorded data not only at episode reset (pressed M) but also if App exit event was issued or all episodes were done and have to exit (pressed ESC).

Reply:
I think we're asking for trouble writing code this way. Let's have a single code-path for "end of episode". I think this could be `_check_reset_environment`. Call `_check_save_episode_data` in there. And call `_check_reset_environment` here when the user presses ESC. And we could consider renaming `_check_reset_environment` to `_end_episode_and_try_reset` or similar.

Reply:
Nice catch, btw! I guess the original code failed to save the last episode.

Reply:
:white_check_mark: